apps/gdalbuildvrt_lib.cpp

```diff
@@ -800,7 +811,8 @@ std::string VRTBuilder::AnalyseRaster(GDALDatasetH hDS,
             {
                 CPLString osExpected = GetProjectionName(pszProjectionRef);
                 CPLString osGot = GetProjectionName(proj);
-                return CPLSPrintf("gdalbuildvrt does not support heterogeneous "
+                return m_osProgramName +
```

Comment:
Is there a reason to do both string concatenation and printf?

Reply:
basically because it is more convenient to write it that way than adding %s and .c_str()